perf/flame.jl

```diff
@@ -37,17 +37,17 @@ ProfileCanvas.html_file(joinpath(output_dir, "flame.html"), results)
 #####
 
 allocs_limit = Dict()
-allocs_limit["flame_perf_target"] = 148_592
-allocs_limit["flame_perf_target_tracers"] = 180_848
+allocs_limit["flame_perf_target"] = 152_672
+allocs_limit["flame_perf_target_tracers"] = 184_928
 allocs_limit["flame_perf_target_edmfx"] = 7_005_552
-allocs_limit["flame_perf_diagnostics"] = 25_356_928
-allocs_limit["flame_perf_target_diagnostic_edmfx"] = 1_311_040
+allocs_limit["flame_perf_diagnostics"] = 26_645_600
+allocs_limit["flame_perf_target_diagnostic_edmfx"] = 1_323_136
 allocs_limit["flame_sphere_baroclinic_wave_rhoe_equilmoist_expvdiff"] =
     4_018_252_656
 allocs_limit["flame_perf_target_threaded"] = 1_276_864
-allocs_limit["flame_perf_target_callbacks"] = 37_277_112
-allocs_limit["flame_perf_gw"] = 3_226_430_064
-allocs_limit["flame_perf_target_prognostic_edmfx_aquaplanet"] = 1_258_848
+allocs_limit["flame_perf_target_callbacks"] = 386_584
```

Comment:
What was the key change that enabled this drop in allocs?

Reply:
The old callbacks computed the diagnostics, and the old diagnostics were allocating a lot.

Reply:
Are we sure that this hasn't just dropped coverage? The old diagnostics were doing a lot of computations, are we just no longer computing those diagnostics?

Reply:
Or is it because the diagnostics are being computed offline now?

Reply:
The old diagnostics had lots of allocation issues that got fixed in the new diagnostics. These are the issues that were discussed in a previous PR (e.g., the `UVector`)